simpleclient/src/main/java/io/prometheus/client/Collector.java

```diff
@@ -150,44 +149,43 @@ public MetricFamilySamples filter(Predicate<String> sampleNameFilter) {
      * {@code # HELP}), and as this name <a href="https://github.com/prometheus/common/issues/319">must be unique</a>
      * we include the name without suffix here as well.
      */
-    public String[] getNames() {
+    public List<String> getNames() {
```

Comment:
What about leaving `getNames()` as it is?

The `_created` names in `getNames()` are used to prevent illegal name clashes, like somebody registering a counter named `my_counter_total` but also a gauge named `my_counter_created`. These are a weird corner cases anyway, and I think it's strange to allow this when the `PROMETHEUS_DISABLE_CREATED_SERIES` environment variable is set.

Moreover, if we leave `getNames()` as it was, `PROMETHEUS_DISABLE_CREATED_SERIES` would just enable/disable the `_created` series. If we change `getNames()`, we will allow weird corner cases where an application breaks unless `PROMETHEUS_DISABLE_CREATED_SERIES` is set. I don't think this is the intention.

Reply:
According to https://github.com/prometheus/client_java/issues/774#issuecomment-1133955882 these are actually _not_ forbidden just discouraged. Sure, evil minded  integrators may abuse this by developing their code under `PROMETHEUS_DISABLE_CREATED_SERIES=true` but it will break on the first time it will _not_ be used.

Reply:
Fair enough - I've reverted the this change. Thanks for taking the time and effort explaining it!